haas/api.py

```diff
@@ -510,6 +510,25 @@ def headnode_detach_network(headnode, hnic):
                             # Network Code #
                             ################
 
+@rest_call('GET', '/networks')
+def list_networks():
+    """List all networks.
+    Returns a JSON dictionary, indexed by the name work names whose values
+    are ditionaries with index as channel and project name.
```

Comment:
Better doc string for the first two lines: 

Returns a JSON dictionary of dictionaries, indexed by the name of the network.
The interior dictionaries are indexed by project and channel.